gateway/ln-gateway/src/client.rs

```diff
@@ -94,9 +96,24 @@ impl GatewayClientBuilder {
             client_builder.with_database(rocksdb);
         }
 
+        let client_secret = match client_builder
```

Comment:
**suggestion:** Looking at this makes me realize that possibly secret should be a part of the builder.

So the whole thing turns into:

```
client_builder
  .secret_new_if_missing(|| PlainRootSecretStrategy::random(...))
  .build()
  .await
  .map_err(GatewayError::ClientStateMachineError)?
```

or something like that.

Reply:
Oh, and if we go something like `.secret_new_if_missing(|| lambda)`, there's nothing preventing us from turning this confusing ZST "strategy` into `trait EncodedSecret` or something.

Reply:
What I didn't like about the strategy pattern was that it was implicit and hidden away whether a secret is being loaded and re-used, or whether a secret is being generated and stored. A function like `.secret_new_if_missing(|| PlainRootSecretStrategy::random(...))` takes us back to that downside in my mind. I like the caller providing explicit instructions and handling both cases externally. What do you think?

Reply:
I guess I just like the superficial natural flow of the builder pattern. When the chain gets broken, especially with a larger block, it becomes less clear what is going on.

With the builder pattern the advantage is that we can just use lambdas and we don't need to deal with traits. 

A

```
builder
  .stmh()
  .smth();

let secret = builder.bar.bar + 12 some calls;

builder.build(secret);
```

Just structurally looks worse than:

```
builder
  .stmh()
  .smth();
  .secret(|prev/builder/somth| smth.bar.bar + 12 some calls)
  .build();
```
  
as the details of dealing with the secret are structurally delegated (extra indentation).

In both cases we did move out dealing with the secrets from the internals of the `build`, giving user more flexibility. It's just looks a bit more readable, IMO.

`secret_new_if_missing` is kind of clunky, but maybe some `secret_with(|existing : Option<_>| ...)` would be handier. Or something.

Not a blocker, we can iterate later.

Reply:
The real question seems to be who is responsible for storing the secret. This PR makes it kinda an external responsibility, but the builder has some helper functions (because it has a DB reference I guess), which feels a bit unclean. In the future there could be a `SecretStore` object/trait that is independent of the client builder to separate concerns fully.

@dpc's approach would keep it a client responsibility with a simple way to override instead.

I think in the long run having an external secret store abstraction is the more flexible solution and we can iterate on developer UX as we go.